TASK-04: Integration with Database

nodm opened the pull request:
Lambda endpoints:
* addProduct: [POST https://jdpcg4ienj.execute-api.us-east-1.amazonaws.com/dev/products](https://jdpcg4ienj.execute-api.us-east-1.amazonaws.com/dev/products)
* getProductById: [GET https://jdpcg4ienj.execute-api.us-east-1.amazonaws.com/dev/products/{id}](https://jdpcg4ienj.execute-api.us-east-1.amazonaws.com/dev/products/66e8e6bd-70f1-4c0b-ae0d-a90961065055)
* getProductList: [GET https://jdpcg4ienj.execute-api.us-east-1.amazonaws.com/dev/products](https://jdpcg4ienj.execute-api.us-east-1.amazonaws.com/dev/products)

Website on Cloudfront
* [My Store!](https://d2oxj55y64zodz.cloudfront.net/)

nodm committed:
TASK-04: Use Product data from the AWS RDS PostrgreSQL

nodm committed:
TASK-04: Close DB connection and improve error handling

nodm committed:
TASK-04: Add lambda function for creating new product

nodm committed:
TASK-04: Remove schema from the requests

nodm committed:
TASK-04: Transform DB field name to the name that used in the model

nodm committed:
TASK-04: Add task 4 description

nodm committed:
TASK-04: Add body validation

nullpr commented:
@nodm hi there, it looks like https://jdpcg4ienj.execute-api.us-east-1.amazonaws.com/dev/products/{id} does not work, I've tried a few ids (eg: 1, 6e8e6bd-70f1-4c0b-ae0d-a90961065055) it always returns internal server error

also, I cannot create new product
<img width="856" alt="image" src="https://user-images.githubusercontent.com/3364804/180957168-7ec19294-688b-4163-9431-d1c6ded81734.png">

from the FE side: customers can add to the basket more products that actually are available in stock

nodm committed:
TASK-04: Fix issue about uuid and string comparison in SQL request

nodm committed:
TASK-04: Add missed 'required' validation rules.

nodm commented:
Hi Taras!
Thank you so much for your comment.

1. I made a mistake in the link to the `getProductById` function in the PR comment. This is now fixed so you can try again. Also, I fixed an issue with uuid and string comparison in the [commit](https://github.com/nodm/shop-react-redux-services/pull/2/commits/d65f719c809f5c4fcf3338df0edfbbee762e9229). So everything looks fine now at least from my perspective.
2. `price` and `count` are mandatory keys that I missed in the validation schema. Fixed in the [commit](https://github.com/nodm/shop-react-redux-services/pull/2/commits/979bb662d235ea39a5656bccee742b181d10e499).
3. Maybe :) I did not modify FE in the scope of Task 4.
